### PR TITLE
Changing msazure.py to be compliant with azure sdk v1.0.2 and adding …

### PR DESCRIPTION
**Pull request by mabsou**
…endpoint for SMB comminucation needed for deploying salt on windows instances.

The previous version was throwing an error when used with azure sdk v0.10 and the smb endpoint is needed to when deploying salt on windows instances or it cannot transfer the salt executable.

### CODE REVIEW

**Comment by jfindlay** [comment]
@mabsou, thanks for the fixes.  There is a [lint error](https://jenkins.saltstack.com/job/salt-pr-lint-n/10541/violations/file/salt/cloud/clouds/msazure.py/) and it looks like you've included an extra file in 161fb34e13675d952a80b0f544bfb9152613f1d1.  Is it possible for you to remove this commit?

**Comment by techhat** [comment]
@mabsou this is extremely helpful. Aside from the lint error, would you mind changing `return {'Error': '...]}` back to `raise`?

File: salt/cloud/clouds/msazure.py [hunk 1]
```diff
@@ -660,9 +674,9 @@ def wait_for_hostname():
         ret['Attached Volumes'] = created
 
     data = show_instance(vm_['name'], call='action')
-    log.info('Created Cloud VM \'{0[name]}\''.format(vm_))
+    log.info('Created Cloud VM {0[name]!r}'.format(vm_))
```
Comment (terminalmage) [inline salt/cloud/clouds/msazure.py]: The repr flag ``!r`` here is not acceptable, please revert all changes which use it.

**Comment by rallytime** [comment]
@mabsou There are still a couple of places here where `return {'Error': '...]}` is being used instead of `raise`. Once those are addressed, we'd love to get this merged in, because this is great!

**Comment by mabsou** [comment]
Are things okey now ? I fail to understand why the three first tests are failing.

**Comment by rallytime** [comment]
@mabsou The test failures are not related. There are a couple of places still returning an `Error` dictionary instead of using a `raise`, but I can grab those really quickly.

**Pull request merged**